client/packages/system/src/Patient/CreatePatientModal/PatientFormTab.tsx

```diff
@@ -1,117 +1,57 @@
-import React, { FC, useMemo } from 'react';
+import React, { FC, useState } from 'react';
 import {
-  BaseDatePickerInput,
-  BufferedTextInput,
-  DateUtils,
-  InputWithLabelRow,
-  Select,
-  Stack,
-  useFormatDateTime,
-  useTranslation,
-} from '@openmsupply-client/common';
-import { Gender, usePatientCreateStore } from '@openmsupply-client/programs';
+  Gender,
+  JsonData,
+  JsonForm,
+  useFormSchema,
+  usePatientCreateStore,
+} from '@openmsupply-client/programs';
 import { PatientPanel } from './PatientPanel';
 
+type Patient = {
+  code?: string;
+  code2?: string;
+  firstName?: string;
+  lastName?: string;
+  dateOfBirth?: string;
+  gender?: Gender;
+};
+
 export const PatientFormTab: FC<PatientPanel> = ({ patient, value }) => {
   const { updatePatient } = usePatientCreateStore();
-  const t = useTranslation('patients');
-
-  const genderOptions = useMemo(
-    () =>
-      Object.keys(Gender).map(key => ({
-        label: key,
-        value: key,
-      })),
-    []
+  const { data: patientCreationUI, isError, isLoading } = useFormSchema.document.byType(
+    'PatientCreationJSONForms'
   );
-  const dateFormatter = useFormatDateTime().customDate;
+  const [data, setData] = useState<Patient | undefined>();
```

Comment:
Found the update bug: You call updatePatient with `data` and not with `newData`. `data` will only be updated in the next render cycle...

Think you don't need the useState at all do you?

Reply:
I put it in a state so that the data can be passed through to the JsonForm component 🤔 

```suggestion
      const patientData = newData as Patient;
      updatePatient({
        code: patientData?.code,
        code2: patientData?.code2,
        firstName: patientData?.firstName,
        lastName: patientData?.lastName,
        dateOfBirth: patientData?.dateOfBirth,
        gender: patientData?.gender,
      });
```
^ having newData as Patient in a variable solves this too...

Reply:
oh yes, you could use the patient from the usePatientCreateStore though...  but there might be some old data in this store(?) so having the useState is fine!

Reply:
[ed60249](https://github.com/openmsupply/open-msupply/pull/1262/commits/ed60249ab0f093d3d723a22e5b17e688682ff032) 

🤔 I put a default patient creation in the FE...